nservicebus/upgrades/asp-saga-pruning.md

```diff
@@ -0,0 +1,55 @@
+---
+title: Azure Storage Persistence upgrade Version 6.2.4 to 6.2.5
+summary: Instructions on how to patch Azure Storage Persistence when orphan saga index records appear.
+reviewed: 2016-06-30
+tags:
+ - upgrade
+related:
+ - nservicebus/sagas
+ - nservicebus/azure-storage-persistence
+---
+
+
+## Summary
+
+This document explains how to upgrade and patch a system for [Azure Storage Persistence bug #74](https://github.com/Particular/NServiceBus.Persistence.AzureStorage/issues/74) using NServiceBus.Azure hotfix release 6.2.5.
+
+WARNING: When upgrading to NServiceBus.Persistence.AzureStorage Version 1 and above, the following upgrade will need to be performed prior to beginning any other upgrade steps.
+
+WARNING: The [saga de-duplication patch](/nservicebus/upgrades/asp-saga-deduplication.md) process must have been completed at least once prior to proceeding with this update.
+
+
+### How to know if a system may be affected
+
+This bug will affect any system that has ever used sagas and NServiceBus.Azure v6.2.4 or lower.
+
+
+## Patch Requirements
+
+To deploy this fix throughout a system, all endpoints will need to be upgraded and saga data that has been stored by the Azure Storage persister will need to be patched.
+
+
+### Upgrading endpoints
+
+All endpoints using NServiceBus.Azure will need to be upgraded to version 6.2.5 or higher.
+
+
+### Patching data
+
+Saga data stored in Azure will need to be patched using the `IndexPruner` utility which can be downloaded from [https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284](https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284). This utility will remove all orphaned secondary indexes from the Azure Storage Tables.
+
+
+## Patch steps
+
+ 1. Download the index pruning tool from [https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284](https://github.com/Particular/IssueDetection/releases/tag/nsb.azure.284) to a computer that has the .NET Framework 4.5.2 installed and also has internet access .
+ 1. The `IndexPruner` utility requires an Azure Storage connection string. The connection string can be added to the `IndexPruner.exe.config` file with the `name="sagas"` value or provided as a commandline parameters (shown in step 4).
+	```xml
+	<configuration>
+		<connectionStrings>
+			<add name="sagas" connectionStrings="--anAzureStorageConnectionString--"/>
+		</connectionStrings>
+	</configuration>
+	```
+ 1. For each endpoint that is being patched, copy all the assemblies that contain saga type type definitions to the same folder as the index pruning tool. These files will be scanned to find all implementations of `IContainSagaData` which will indicate the sagas that need to be pruned in Azure Storage.
```

Comment:
How about:
Copy the endpoint dll along with the assemblies that contain saga type type definitions to the same folder as the index pruning tool. If multiple endpoints require this patch, it is ok to add the assemblies for all the endpoints to the same location. This saves time from having to run the tool for each endpoint. These assemblies will be scanned to find all implementations of `IContainSagaData` which will indicate the sagas that need to be pruned in Azure Storage.